### PR TITLE
#372 - dev_setup.sh ignores WORKON_HOME when VIRTUALENV_ROOT is unset

### LINKED ISSUE #372: dev_setup.sh ignores WORKON_HOME when VIRTUALENV_ROOT is unset
Working with a shell configured to use virtualenv via virtualenvwrapper may result in the creation of a new virtualenv root directory.

For example if WORKON_HOME is set to $HOME/projects, executing dev_setup.sh will create the $HOME/.virtualenv and use it as the location for the "mycroft" virtual environment.

Since virtualenv's documentation does not describe the use of the VIRTUALENV_ROOT variable, I would like to suggest evaluating if WORKON_HOME is not empty before using VIRTUALENV_ROOT, in which case its value becomes used over VIRTUALENV_ROOT's default.

### PR DESCRIPTION
**Pull request by ncouture**
```
  - favor using the WORKON_HOME environment variable over
    VIRTUALENV_ROOT
  - closes #372 
```

### CODE REVIEW

**Comment by ncouture** [comment]
These changes are included in PR #378.

File: dev_setup.sh [hunk 1]
```diff
@@ -33,7 +33,12 @@ if [ $(id -u) -eq 0 ]; then
 fi
 
 TOP=$(cd $(dirname $0) && pwd -L)
-VIRTUALENV_ROOT=${VIRTUALENV_ROOT:-"${HOME}/.virtualenvs/mycroft"}
+
+if [ -z "$WORKON_HOME" ]; then
+    VIRTUALENV_ROOT=${VIRTUALENV_ROOT:-"${HOME}/.virtualenvs/mycroft"}
+else
+    VIRTUALENV_ROOT="$WORKON_HOME"
```
Comment (jdorleans) [inline dev_setup.sh]: It appears to me we should concatenate WORKON_HOME + /mycroft

`VIRTUALENV_ROOT="$WORKON_HOME/mycroft"`

**Comment by ncouture** [comment]
@jdorleans see last commit

**Comment by aatchison** [comment]
We have made a major update to master and plan on getting to this, thanks!

**Pull request merged**